engine/schema/src/org/apache/cloudstack/outofbandmanagement/dao/OutOfBandManagementDaoImpl.java

```diff
@@ -0,0 +1,163 @@
+
+@DB
+@Component
+@Local(value = {OutOfBandManagementDao.class})
+public class OutOfBandManagementDaoImpl extends GenericDaoBase<OutOfBandManagementVO, Long> implements OutOfBandManagementDao {
+    private static final Logger LOG = Logger.getLogger(OutOfBandManagementDaoImpl.class);
+
+    private SearchBuilder<OutOfBandManagementVO> HostSearch;
+    private SearchBuilder<OutOfBandManagementVO> ManagementServerSearch;
+    private SearchBuilder<OutOfBandManagementVO> OutOfBandManagementOwnerSearch;
+    private SearchBuilder<OutOfBandManagementVO> StateUpdateSearch;
+
+    private Attribute PowerStateAttr;
+    private Attribute MsIdAttr;
+    private Attribute UpdateTimeAttr;
+
+    public OutOfBandManagementDaoImpl() {
+        super();
+
+        HostSearch = createSearchBuilder();
+        HostSearch.and("hostId", HostSearch.entity().getHostId(), SearchCriteria.Op.EQ);
+        HostSearch.done();
+
+        ManagementServerSearch = createSearchBuilder();
+        ManagementServerSearch.and("server", ManagementServerSearch.entity().getManagementServerId(), SearchCriteria.Op.EQ);
+        ManagementServerSearch.done();
+
+        OutOfBandManagementOwnerSearch = createSearchBuilder();
+        OutOfBandManagementOwnerSearch.and("server", OutOfBandManagementOwnerSearch.entity().getManagementServerId(), SearchCriteria.Op.EQ);
+        OutOfBandManagementOwnerSearch.or("serverNull", OutOfBandManagementOwnerSearch.entity().getManagementServerId(), SearchCriteria.Op.NULL);
+        OutOfBandManagementOwnerSearch.done();
+
+        StateUpdateSearch = createSearchBuilder();
+        StateUpdateSearch.and("status", StateUpdateSearch.entity().getPowerState(), SearchCriteria.Op.EQ);
+        StateUpdateSearch.and("id", StateUpdateSearch.entity().getId(), SearchCriteria.Op.EQ);
+        StateUpdateSearch.and("update", StateUpdateSearch.entity().getUpdateCount(), SearchCriteria.Op.EQ);
+        StateUpdateSearch.done();
+
+        PowerStateAttr = _allAttributes.get("powerState");
+        MsIdAttr = _allAttributes.get("managementServerId");
+        UpdateTimeAttr = _allAttributes.get("updateTime");
+        assert (PowerStateAttr != null && MsIdAttr != null && UpdateTimeAttr != null) : "Couldn't find one of these attributes";
+    }
+
+    @Override
+    public OutOfBandManagement findByHost(long hostId) {
+        SearchCriteria<OutOfBandManagementVO> sc = HostSearch.create("hostId", hostId);
+        return findOneBy(sc);
+    }
+
+    @Override
+    public List<OutOfBandManagementVO> findAllByManagementServer(long serverId) {
+        SearchCriteria<OutOfBandManagementVO> sc = OutOfBandManagementOwnerSearch.create();
+        sc.setParameters("server", serverId);
+        return listBy(sc, new Filter(OutOfBandManagementVO.class, "updateTime", true, null, null));
+    }
+
+    private void executeExpireOwnershipSql(final String sql, long resource) {
+        TransactionLegacy txn = TransactionLegacy.currentTxn();
+        try {
```

Comment:
To ensure that the transaction is properly closed, declare as a try with resources.